app/services/recommendation_service.py

```
@@ -3,82 +3,361 @@
 from typing import List, Dict, Any, Optional, Tuple
 from datetime import datetime
 from sqlalchemy.ext.asyncio import AsyncSession
-from sqlalchemy import select
+from sqlalchemy import select, and_, or_, not_
```

Ruff (0.11.9)

`sqlalchemy.or_` imported but unused

Remove unused import: `sqlalchemy.or_`

(F401)